daemon/session_agent.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/gorilla/mux"
+	"gopkg.in/tomb.v2"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+type SessionAgent struct {
+	Version  string
+	listener net.Listener
+	serve    *shutdownServer
+	tomb     tomb.Tomb
+	router   *mux.Router
+
+	mu sync.Mutex
+}
+
+func (s *SessionAgent) Init() error {
+	listenerMap, err := activationListeners()
+	if err != nil {
+		return err
+	}
+	agentSocket := fmt.Sprintf("%s/%d/snap-session.socket", dirs.XdgRuntimeDirBase, os.Getuid())
+	if listener, err := getListener(agentSocket, listenerMap); err == nil {
+		s.listener = &ucrednetListener{listener}
+	} else {
+		return fmt.Errorf("when trying to listen on %s: %v", agentSocket, err)
```

Comment:
We're trying to use error messages that follow the pattern `cannot ...`, so perhaps:
```suggestion
		return fmt.Errorf("cannot listen on socket %s: %v", agentSocket, err)
```